tests/unit/test_webhook.py

```diff
@@ -527,18 +568,51 @@ def test_patch_acceptable_issue_problem(self, mock_proxy):
             'X-Hub-Signature': signature,
         }
         with webcompat.app.test_client() as c:
-            mock_proxy.return_value.status_code = 400
+            mock_moderate.side_effect = HTTPError()
+            mock_moderate.status_code = 400
             rv = c.post(
                 '/webhooks/labeler',
                 data=json_event,
                 headers=headers
             )
-            self.assertEqual(rv.data, b'ooops')
-            self.assertEqual(rv.status_code, 400)
-            self.assertEqual(rv.content_type, 'text/plain')
+            assert not mock_close.called, 'close_private_issue should not be called'  # noqa
+            assert rv.data == b'oops'
+            assert rv.status_code == 400
+            assert rv.content_type == 'text/plain'
 
-    @patch('webcompat.webhooks.helpers.private_issue_rejected')
-    def test_patch_not_acceptable_issue(self, mock_proxy):
+    @patch.object(webcompat.webhooks.model.WebHookIssue,
+                  'reject_private_issue')
+    @patch.object(webcompat.webhooks.model.WebHookIssue, 'close_private_issue')
+    def test_patch_acceptable_issue_problem(self, mock_close, mock_reject):
```

Comment:
Probably not testing what we want. 

There are two parts for these tests.

1. Testing the functions which are in the modules. (easy)
2. Testing the HTTP interactions outcome (aka the thing in the `__init__` which receives the HTTP requests `c.post`)

Probably everything else doesn't really need a POST request to `/webhooks/labeler`. It's more what is the input function and what is the output of the function.

Let's take `tag_as_public`.

1. This is a function. It creates a payload request and answer a response through make_request. 
    - The patch needs to be only about `make_requests` 
    - the tests needs to cover do we send the right payload.
    - The HTTP response variability does NOT really affect the outcome at this level.
2. `tag_as_public` as used in `process_issue_action` 
    - The patch needs to be only about `tag_as_public`
    - the test needs to cover the different HTTP response.
    - the HTTP responses is what create variations here on process_issue_action. So basically the input is an `issue`, this issue depending on its inner criteria will trigger `tag_as_public` which can have two different responses.
   

When thought like this for each part it becomes a lot easier to create all tests. The `c.post`  should only be for testing the route and this happens only in __init__.py. Basically you inherited from **bad tests from me** 😝 

And trying to convert them to the new model doesn't fix the bad.

Reply:
Mayyyyybe deleting those tests (or simplifying them) can be what happens in #3436 (or we just close that one).